### DISCUSSION TITLE
Install Laravel Lang Packages, in `--dev` mode?

### DISCUSSION THREAD

**Discussion by luisprmat** [Laravel-Lang/common#201] [Q&A] [answered]
This project has grown quite a bit and now has numerous very useful packages. The specific question is: the package `laravel-lang/common` is 
*recommended* to be installed in `--dev` mode and it contains the other packages (like `models` ,`routes` or `locales`) **BUT** If I want to use package `laravel-lang/routes` or `laravel-lang/models` in production, for example, should I install it separately without `--dev` mode? 

How would I avoid having unnecessary code in production and not have problems due to missing packages? This is because in `production` you install `composer install --no-dev`.

**Comment by andrey-helldar** [comment, marked as answer]
Indeed, when packages were scarce, the bulk of them were destined for the `dev` area.

For example, to be able to use the facade in production to get a list of installed localizations, you had to do the following:

```bash
composer require laravel-lang/common --dev
composer require laravel-lang/locales
```

But now the number of packages useful for production has outgrown the number of packages, and it turns out that the documentation recommends doing so:

```bash
composer require laravel-lang/common --dev
composer require laravel-lang/locales laravel-lang/models laravel-lang/routes laravel-lang/json-fallback
```

That is, it offers to manually install 4 dependencies in the `production` area while leaving 5 packages in the `dev` area, which is almost half.

I agree that it is time to remove the recommendation of the `--dev` parameter when installing a `Common` package. I will do so in the near future.

**Reply by andrey-helldar** [reply]
Ready: https://laravel-lang.com/packages-common.html
